doc/sphinxext/gallery_order.py

```diff
@@ -0,0 +1,82 @@
+"""
+Configuration for the order of gallery sections and examples.
+Paths are relative to the conf.py file.
+
+"""
+
+from sphinx_gallery.sorting import ExplicitOrder
+
+# Gallery sections shall be diplayed in the following order.
+# Non-matching sections are appended.
+explicit_order_folders = [
+                          '../examples/api',
+                          '../examples/pyplots',
+                          '../examples/subplots_axes_and_figures',
+                          '../examples/color',
+                          '../examples/statistics',
+                          '../examples/lines_bars_and_markers',
+                          '../examples/images_contours_and_fields',
+                          '../examples/shapes_and_collections',
+                          '../examples/text_labels_and_annotations',
+                          '../examples/pie_and_polar_charts',
+                          '../examples/style_sheets',
+                          '../examples/axes_grid1',
+                          '../examples/axisartist',
+                          '../examples/showcase',
+                          '../tutorials/introductory',
+                          '../tutorials/intermediate',
+                          '../tutorials/advanced']
+
+
+class MplExplicitOrder(ExplicitOrder):
+    """ for use within the 'subsection_order' key"""
+    def __call__(self, item):
+        if item in self.ordered_list:
+            return "{:04d}".format(self.ordered_list.index(item))
+        else:
+            return "zzz" + item
+
+
+# Subsection order:
+# Subsections are ordered by filename, unless they appear in the following
+# lists in which case the list order determines the order within the section.
+# Examples/tutorials that do not appear in a list will be appended.
+
+# Tutorials:
+list_introductory = ["usage", "pyplot", "sample_plots", "images", "lifecycle",
```

Comment:
Constructing all the small lists seems a bit cumbersome and is harder to read because of the variing list positions. I would just do:

~~~
explicit_subsection_order = [item + ".py" for item in [
    #  **Tutorials**
    #  introductory
    "usage", "pyplot", "sample_plots", "images", "lifecycle", "customizing",
    #  intermediate
    "artists", "legend_guide", "color_cycle", "gridspec",
    "constrainedlayout_guide", "tight_layout_guide",
    #  advanced
    #  text
    "text_intro", "text_props",
    #  colors
    "colors",

    #  **Examples**
    #  color
    "color_demo",
    #  pies
    "pie_features", "pie_demo2",
]]
~~~